pyart/graph/max_cappi.py

```diff
@@ -0,0 +1,443 @@
+"""
+Plot Max-CAPPI
+
+This module provides a function to plot a Maximum Constant Altitude Plan Position Indicator (Max-CAPPI)
+from radar data using an xarray dataset. The function includes options for adding map features, range rings,
+color bars, and customized visual settings.
+
+Author: Syed Hamid Ali (@syedhamidali)
+"""
+
+__all__ = ["plot_maxcappi"]
+
+import os
+import warnings
+
+import cartopy.crs as ccrs
+import cartopy.feature as feat
+import matplotlib.pyplot as plt
+import numpy as np
+from cartopy.mpl.gridliner import LATITUDE_FORMATTER, LONGITUDE_FORMATTER
+from matplotlib.ticker import NullFormatter
+
+warnings.filterwarnings("ignore")
+
+
+def plot_maxcappi(
+    grid,
+    field,
+    cmap=None,
+    vmin=None,
+    vmax=None,
+    title=None,
+    lat_lines=None,
+    lon_lines=None,
+    add_map=True,
+    projection=None,
+    colorbar=True,
+    range_rings=False,
+    dpi=100,
+    show_progress=False,
+    savedir=None,
+    show_figure=True,
+    add_slogan=False,
+    **kwargs,
+):
+    """
+    Plots a Constant Altitude Plan Position Indicator (CAPPI) using an xarray Dataset.
+
+    Parameters
+    ----------
+    grid : pyart.core.Grid
+        The grid object containing the radar data to be plotted.
+    field : str
+        The radar field to be plotted (e.g., "REF", "VEL", "WIDTH").
+    cmap : str or matplotlib colormap, optional
+        Colormap to use for the plot. Default is "SyedSpectral" if available, otherwise "HomeyerRainbow".
+    vmin : float, optional
+        Minimum value for the color scaling. Default is None, which sets it to the minimum value of the data.
+    vmax : float, optional
+        Maximum value for the color scaling. Default is None, which sets it to the maximum value of the data.
+    title : str, optional
+        Title of the plot. If None, the title is set to "Max-{field}".
+    lat_lines : array-like, optional
+        Latitude lines to be included in the plot. Default is calculated based on dataset coordinates.
+    lon_lines : array-like, optional
+        Longitude lines to be included in the plot. Default is calculated based on dataset coordinates.
+    add_map : bool, optional
+        Whether to include a map background in the plot. Default is True.
+    projection : cartopy.crs.Projection, optional
+        The map projection for the plot. Default is cartopy.crs.LambertAzimuthalEqualArea().
+    colorbar : bool, optional
+        Whether to include a colorbar in the plot. Default is True.
+    range_rings : bool, optional
+        Whether to include range rings at 50 km intervals. Default is True.
+    dpi : int, optional
+        DPI (dots per inch) for the plot. Default is 100.
+    show_progress : bool, optional
+        Whether to print progress messages. Default is False.
+    savedir : str, optional
+        Directory where the plot will be saved. If None, the plot is not saved.
+    show_figure : bool, optional
```

Comment:
Same here, maybe leave it up to the use for plt.show()?

Reply:
Actually the it doesn't take the fig and ax arguments since its fig canvas (Main, upper, right and colorbar axes) is manually built, that is why. What do you suggest?